src/main/java/Duke.java

```diff
@@ -1,10 +1,107 @@
+import java.util.Scanner;
+
 public class Duke {
+
+    private static Task[] tasks = new Task[100];
+    private static Scanner sc = new Scanner(System.in);
+
+    public static void listTasks() {
+        if (Task.totalTasks == 0) {
+            System.out.println("No tasks yet!");
+        } else {
+            for (int i = 0; i < Task.totalTasks; i++) {
+                System.out.printf("%d.%s\n", i+1, tasks[i].toString());
+            }
+        }
+    }
+
+    public static void markTasksAsDone(String input) {
+        String[] inputArray = input.split(" ");
+
+        //completedIndex holds the index of valid integer(s) in inputArray (indicating index in tasklist)
+        int completedIndex;
+        for (String word: inputArray) {
+            if (word.equals("done")) {
```

Comment:
Maybe you can consider splitting this if else statements as they seem very nested.